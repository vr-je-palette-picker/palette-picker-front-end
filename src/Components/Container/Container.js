import React, { Component } from 'react';
import HomeNav from '../HomeNav/HomeNav.js';
import ColorCard from './../ColorCard/ColorCard.js';
import './Container.scss';

class Container extends Component {
  constructor() {
    super();
    this.state = {
      color_1: {
        hex_code: '',
        locked: false
      },
      color_2: {
        hex_code: '',
        locked: false
      },
      color_3: {
        hex_code: '',
        locked: false
      },
      color_4: {
        hex_code: '',
        locked: false
      },
      color_5: {
        hex_code: '',
        locked: false
      }
    };
  }

  componentDidMount = () => {
    this.generateRandomColors();
  };

  generateRandomHex = () => {
    return '#' + Math.floor(Math.random() * 16777215).toString(16);
  };

  generateRandomColors = () => {
    for (let color of ['color_1', 'color_2', 'color_3', 'color_4', 'color_5']) {
      if (!this.state[color].locked) {
        this.setState({
          [color]: {
            hex_code: this.generateRandomHex(),
            locked: false
          }
        });
      }
    }
  };

  generateColorCards = colors => {
    return colors.map((color, index) => {
      const { hex_code, locked } = color;
      return (
        <ColorCard
          key={index}
          color={`color_${index + 1}`}
          hex={hex_code}
          locked={locked}
          toggleLock={this.toggleLock}
        />
      );
    });
  };

  toggleLock = (hex, color) => {
    this.setState({
      [color]: {
        hex_code: hex,
        locked: !this.state[color].locked
      }
    });
  };

  render() {
    return (
      <main className='Container'>
        <HomeNav
          randomColors={this.generateRandomColors}
        />
        <section className='Container__section'>
          {this.generateColorCards([
            this.state.color_1,
            this.state.color_2,
            this.state.color_3,
            this.state.color_4,
            this.state.color_5
          ])}
        </section>
      </main>
    );
  }
}

export default Container;
