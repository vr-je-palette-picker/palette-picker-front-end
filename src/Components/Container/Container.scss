.Container {
  display: flex;
  height: 80vh;
  width: 100vw;
}

.Container__section {
  display: flex;
}
