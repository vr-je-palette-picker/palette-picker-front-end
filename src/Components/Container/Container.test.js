import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  let wrapper;
  let hexCodes = ['', '', '', '', ''];

  beforeEach(() => {
    wrapper = shallow(<Container hexCodes={hexCodes} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
