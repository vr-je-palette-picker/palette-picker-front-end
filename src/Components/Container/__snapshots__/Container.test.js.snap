// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<main
  className="Container"
>
  <HomeNav
    randomColors={[Function]}
    showModal={[Function]}
  />
  <section
    className="Container__section"
  >
    <ColorCard
      color="color_1"
      hex="#F709A7"
      key="0"
      locked={false}
      toggleLock={[Function]}
    />
    <ColorCard
      color="color_2"
      hex="#EF0DEC"
      key="1"
      locked={false}
      toggleLock={[Function]}
    />
    <ColorCard
      color="color_3"
      hex="#B04D8E"
      key="2"
      locked={false}
      toggleLock={[Function]}
    />
    <ColorCard
      color="color_4"
      hex="#ABBDFF"
      key="3"
      locked={false}
      toggleLock={[Function]}
    />
    <ColorCard
      color="color_5"
      hex="#13142B"
      key="4"
      locked={false}
      toggleLock={[Function]}
    />
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="PaletteForm"
      closeTimeoutMS={0}
      isOpen={false}
      onRequestClose={[Function]}
      overlayClassName="PaletteFormOverlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": 100,
            "left": 300,
            "position": "fixed",
            "right": 300,
            "top": 100,
          },
          "overlay": Object {
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          },
        }
      }
    >
      <PaletteForm
        close={[Function]}
        findProjectByName={[Function]}
        newPalette={[Function]}
      />
    </Modal>
  </section>
</main>
`;
